Yoast/wordpress-seo
# Add tests for FAQ and breadcrumbs Schema generators

## Context


* We want to test every code path within the FAQ Schema generator and the breadcrumb schema generator to be able to check whether the implementation is correct.
* This PR also fixes some bugs in the FAQ Schema code:
  * A wrong `numberOfItems` was generated.
  * Two identical `acceptedAnswer` properties were generated and added to each question in the FAQ Schema.

## Summary


This PR can be summarized in the following changelog entry:

* Adds tests to FAQ Schema generator and the breadcrumbs schema generator.
* Fixes a bug where two identical `acceptedAnswer` properties were generated for a  question in the FAQ Schema.
* Fixes a bug where a wrong question count was generated for the FAQ Schema.
* Fixes a bug where an empty URL in a breadcrumb fell back to the canonical of the current page for intermediate breadcrumbs.
* Fixes a bug where an empty text in a breadcrumb fell back to the title of the current page for intermediate breadcrumbs.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Run the unit tests with code coverage enabled.
* Check that the code coverage for `src/generators/schema/breadcrumb.php` and `src/generators/schema/faq.php` is 100%.

## UI changes

## Documentation

## Quality assurance


Fixes #
